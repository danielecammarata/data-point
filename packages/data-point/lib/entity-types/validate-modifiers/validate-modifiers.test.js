/* eslint-env jest */

const {
  validateProperties,
  validateModifiers
} = require("./validate-modifiers");

describe("validateProperties", () => {
  test("It should throw error if keys do not match", () => {
    expect(() => {
      const spec = {
        foo: true
      };
      validateProperties("foo:test", spec, ["bar"]);
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      const spec = {
        bar: true,
        foo: true
      };
      validateProperties("foo:test", spec, ["bar"]);
    }).toThrowErrorMatchingSnapshot();
  });

  test("It should do nothing if using all valid keys", () => {
    let spec = {
      bar: true
    };
    expect(validateProperties("foo:test", spec, ["bar"])).toBeTruthy();

    spec = {};
    expect(validateProperties("foo:test", spec, ["bar"])).toBeTruthy();
  });
});

describe("validateModifiers", () => {
  test("It should include base modifiers", () => {
    expect(() => {
      const spec = {
        inputType: true,
        outputType: true,
        before: true,
        after: true,
        error: true,
        value: true,
        foo: true
      };
      validateModifiers("foo:test", spec, ["bar"]);
    }).toThrowErrorMatchingSnapshot();
  });
});
