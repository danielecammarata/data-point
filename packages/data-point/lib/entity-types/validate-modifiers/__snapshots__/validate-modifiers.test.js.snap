// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validateModifiers It should include base modifiers 1`] = `
"Entity \\"foo:test\\" did not recognize the following properties:
 foo
Valid properties for this entity are:
 inputType, before, value, after, outputType, error, params, bar
Please review your entity and make any necessary corrections so it can be parsed:
'foo:test': {
  \\"inputType\\": true,
  \\"outputType\\": true,
  \\"before\\": true,
  \\"after\\": true,
  \\"error\\": true,
  \\"value\\": true,
  \\"foo\\": true
}"
`;

exports[`validateProperties It should throw error if keys do not match 1`] = `
"Entity \\"foo:test\\" did not recognize the following properties:
 foo
Valid properties for this entity are:
 bar
Please review your entity and make any necessary corrections so it can be parsed:
'foo:test': {
  \\"foo\\": true
}"
`;

exports[`validateProperties It should throw error if keys do not match 2`] = `
"Entity \\"foo:test\\" did not recognize the following properties:
 foo
Valid properties for this entity are:
 bar
Please review your entity and make any necessary corrections so it can be parsed:
'foo:test': {
  \\"bar\\": true,
  \\"foo\\": true
}"
`;
